src/Frame.cpp

```diff
@@ -759,11 +759,11 @@ void Frame::AddColor(int new_width, int new_height, std::string new_color)
 }
 
 // Add (or replace) pixel data to the frame
-void Frame::AddImage(int new_width, int new_height, int bytes_per_pixel, QImage::Format type, const unsigned char *pixels_)
+void Frame::AddImage(int new_width, int new_height, int bytes_per_pixel, int bytes_per_line, QImage::Format type, const unsigned char *pixels_)
 {
 	// Create new buffer
 	const GenericScopedLock<juce::CriticalSection> lock(addingImageSection);
-	int buffer_size = new_width * new_height * bytes_per_pixel;
+	int buffer_size = bytes_per_line * new_height;
```

Comment:
Alternatively, if you _are_ gonna change the signature of `Frame::AddImage()`, it makes sense to _**drop**_ `bytes_per_pixel` (and replace it with `bytes_per_line`), since it no longer has **any** purpose.

(Plus it's totally implied by `QImage::Format`, so it was never necessary to pass it in anyway.)